### ISSUE TITLE
Define npm scripts

### ISSUE THREAD

**Issue by alanmoo** [MozillaFoundation/mofo-standards#19]
Obviously we've got `start`, but what about development scripts? `npm run dev`?

**Comment by gvn** [comment]
By "development script", do you mean running a local build?

**Comment by alanmoo** [comment]
Yup, scripts for use during local development. I feel like we should have at least one standard one across projects which perhaps runs an install and watches files when run.

**Comment by gvn** [comment]
Usually that's already covered by `start`.

I'd standardize `start` as containing:
- `npm i`
- Compile code
- Run server(s) and watch tasks in parallel
- Any other tasks needed for local dev
- Launch app (?)

**Comment by gvn** [comment]
Proposed standard naming:
- `start`
- `build:*`
- `server:*`
- `watch:*`
- `test:*`
- - with the `:*` denoting potential subtasks, eg `watch:html`.

**Comment by alanmoo** [comment]
I'm still not sure I love `npm i` being a part of `start`\- it often slows things down. If we started using yarn, I could see that as it's pretty fast to validate existing packages are installed properly. Though I feel like I've run into [this](https://github.com/npm/npm/issues/10343) issue before. 

Also, would `server` then be the command to run the project as it will run in production? Basically `start` without the watch tasks? I feel like we should go the other way where `start` works as production, but then a `dev` task sprinkles in some extra usefulness (like `watch`).

**Comment by gvn** [comment]
> I'm still not sure I love npm i being a part of start- it often slows things down. 

Oh, I thought you liked it because you asked for it on Science. I'm fine not having `npm i` as part of `start`, but I don't find it problematic either. I'm not generally re-running `start` repeatedly. `yarn` is a good reason to leave it out if people want to use it.

> Also, would server then be the command to run the project as it will run in production?

I like to do either: 
- Just `server` if dev and prod have no real difference. (EG: static generated sites)
- `server:dev` and `server:prod` if there's a difference. (EG: `science.mozilla.org`)

>  I feel like we should go the other way where start works as production, but then a dev task sprinkles in some extra usefulness (like watch).

That's the opposite of the convention that we've been using on most projects.
